### PR TITLE
correct firmware naming #1933

### PR DESCRIPTION
**Pull request by proddy**
See https://github.com/emsesp/EMS-ESP32/issues/1933 for the description.

New naming is

`EMS-ESP-<version>-<chip type>-<flash mem>[+].bin`

`chip type` is as defined in esp_platform from pio. `ESP32`, `ESP32S3`, `ESP32C3` etc. All uppercase.
`flash mem` is either `4MB` or `16MB`.
`+` is added at the end if there is PSRAM

### CODE REVIEW

File: platformio.ini [hunk 1]
```diff
@@ -81,31 +81,33 @@ lib_ldf_mode = chain+
...
-board_build.extra_flags = -DBOARD_HAS_PSRAM
+build_flags =
+  ${espressi32_base_tasmota.build_flags}
+  '-DEMSESP_DEFAULT_BOARD_PROFILE="S32"'
 
-[env:ci_s3]
-; 16MB ESP32-S3 - like BBQKees S3
+[env:ci_s_16M]
+; 16MB ESP32 - no PSRAM - like some of the older BBQKees S32
 extends = espressi32_base
```
Comment (MichaelDvP) [inline platformio.ini:101]: without psram it should use the tasmota base to save ram.

Comment (proddy) [reply platformio.ini]: good idea. I modified it.

File: platformio.ini [hunk 2]
```diff
@@ -81,31 +81,33 @@ lib_ldf_mode = chain+
...
+[env:ci_s_4M]
+; 4MB ESP32 - using Tasmota - no SSL, no PSRAM - like the BBQKees older S32 models
 extends = espressi32_base_tasmota
 extra_scripts = scripts/rename_fw.py
 board = esp32dev
 board_build.partitions = esp32_partition_4M.csv
-board_build.extra_flags = -DBOARD_HAS_PSRAM
+build_flags =
+  ${espressi32_base_tasmota.build_flags}
+  '-DEMSESP_DEFAULT_BOARD_PROFILE="S32"'
```
Comment (MichaelDvP) [inline platformio.ini:97]: ci_s_4M could be S32 or E32 gateway, there should not be a defined DEFAULT_BOARD_PROFILE to trgger the automatic detection of E32 ethernet.

Comment (proddy) [reply platformio.ini:97]: good spot. I'll change!

**Pull request merged**